diff --git a/src/dexed.h b/src/dexed.h
index 8d91150..eb7caf2 100644
--- a/src/dexed.h
+++ b/src/dexed.h
@@ -47,6 +47,9 @@
 #include "EngineMsfa.h"
 #include "EngineMkI.h"
 #include "EngineOpl.h"
+#ifdef RASPBERRYPI_PICO
+#include "dexed_if_common.h"
+#endif
 
 #define NUM_VOICE_PARAMETERS 156
 
@@ -272,6 +275,7 @@ class Dexed
     float getFilterResonance(void);
     void setGain(float gain);
     float getGain(void);
+    void setGain(uint8_t gain);
 
     // Voice configuration methods
     void setOPRateAll(uint8_t rate);
@@ -382,9 +386,12 @@ class Dexed
     uint8_t velocity_offset;
     uint8_t velocity_max;
     float velocity_diff;
+    uint8_t simple_gain=0;
 #ifndef TEENSYDUINO
+#ifndef RASPBERRYPI_PICO
     Compressor* compressor;
 #endif
+#endif
 };
 
 #endif
